aws/table_aws_codestar_notification_rule.go

```diff
@@ -0,0 +1,193 @@
+			{
+				Name:        "targets",
+				Description: "A list of targets associated with the notification rule.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     describeNotificationRule,
+			},
+			{
+				Name:        "created_by",
+				Description: "The name or email alias of the person who created the notification rule.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     describeNotificationRule,
+			},
+			{
+				Name:        "created_timestamp",
+				Description: "The date and time the notification rule was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     describeNotificationRule,
+			},
+			{
+				Name:        "last_modified_timestamp",
+				Description: "The date and time the notification rule was most recently updated.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     describeNotificationRule,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     describeNotificationRule,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     describeNotificationRule,
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     describeNotificationRule,
+				Transform:   transform.FromField("Arn").Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listNotificationRules(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Get client
+	svc, err := CodeStarNotificationsClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_codestar_notification_rule.listNotificationRules", "get_client_error", err)
+		return nil, err
+	}
+
+	params := &codestarnotifications.ListNotificationRulesInput{}
+	paginator := codestarnotifications.NewListNotificationRulesPaginator(svc, params, func(o *codestarnotifications.ListNotificationRulesPaginatorOptions) {
+		o.StopOnDuplicateToken = true
+	})
+
+	for paginator.HasMorePages() {
+		// apply rate limiting
+		d.WaitForListRateLimit(ctx)
+
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_codestar_notification_rule.listNotificationRules", "api_error", err)
+			return nil, err
+		}
+		for _, rule := range output.NotificationRules {
+			d.StreamListItem(ctx, &codestarnotifications.DescribeNotificationRuleOutput{
+				Arn: rule.Arn,
+			})
+			// Context may get cancelled due to manual cancellation or if the limit has been reached
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func describeNotificationRule(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

Comment:
```suggestion
func getCodeStarNotificationRule(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

Please update the function naming wherever necessary.